pkg/cloud/provider/csvprovider_test.go

```diff
@@ -0,0 +1,305 @@
+			GPUClassPricing:  map[string]*price{},
+			GPUMapFields:     []string{},
+			UsesRegion:       true,
+		}
+		key := &csvKey{
+			Labels: map[string]string{"beta.kubernetes.io/arch": "arm64",
+				"beta.kubernetes.io/instance-type":                        "c5.4xlarge",
+				"beta.kubernetes.io/os":                                   "linux",
+				"instance-group":                                          "aorjoa-1",
+				"kubernetes.io/arch":                                      "arm64",
+				"kubernetes.io/hostname":                                  "kind-control-plane",
+				"kubernetes.io/os":                                        "linux",
+				"node-role.kubernetes.io/control-plane":                   "",
+				"node.kubernetes.io/exclude-from-external-load-balancers": "",
+				"providerID":                                              "kind://docker/kind/kind-control-plane",
+				"topology.kubernetes.io/region":                           "us-east-1"},
+			ProviderID: "us-east-1,aorjoa-1",
+			GPULabel:   []string{},
+			GPU:        0}
+
+		node, _, _ := csv.NodePricing(key)
+
+		assertNode(t, node, &models.Node{Cost: "0.60000", PricingType: models.CsvExact})
+	})
+
+	t.Run("it should lookup from ProviderID and Region (NO1. studentID)", func(t *testing.T) {
+		pr := map[string]*price{
+			"us-east-1,aorjoa-1": {
+				EndTimestamp:      "2023-07-24 16:00:00 UTC",
+				InstanceID:        "aorjoa-1",
+				Region:            "us-east-1",
+				AssetClass:        "node",
+				InstanceIDField:   "metadata.labels.instance-group",
+				InstanceType:      "c5.4xlarge",
+				MarketPriceHourly: "0.60000",
+				Version:           "",
+			},
+		}
+
+		key := &csvKey{
+			ProviderID: "us-east-1,aorjoa-1",
+			Labels: map[string]string{
+				"beta.kubernetes.io/instance-type": "c5.4xlarge",
+				"topology.kubernetes.io/region":    "us-east-1",
+			},
+		}
+
+		csv := &CSVProvider{
+			Pricing: pr,
+			NodeClassPricing: map[string]float64{
+				"us-east-1,c5.4xlarge,node": 0.3599,
+			},
+		}
+
+		node, _, _ := csv.NodePricing(key)
+
+		assertNode(t, node, &models.Node{Cost: "0.60000", PricingType: models.CsvExact})
+	})
+
+	t.Run("Node Pricing lookup by region,instance-group key.ID() (us-east-1,aorjoa-1)", func(t *testing.T) {
+		pr := map[string]*price{
+			"us-east-1,aorjoa-1": {
+				EndTimestamp:      "2023-07-24 16:00:00 UTC",
+				InstanceID:        "aorjoa-1",
+				Region:            "us-east-1",
+				AssetClass:        "node",
+				InstanceIDField:   "metadata.labels.instance-group",
+				InstanceType:      "c5.4xlarge",
+				MarketPriceHourly: "0.60001",
+				Version:           "",
+			},
+		}
+
+		key := &csvKey{
+			ProviderID: "us-east-1,aorjoa-1",
+		}
+
+		csv := &CSVProvider{
+			Pricing: pr,
+		}
+
+		node, _, _ := csv.NodePricing(key)
+
+		assertNode(t, node, &models.Node{Cost: "0.60001", PricingType: models.CsvExact})
+	})
+
+	t.Run("Try without a region to be sure key.ID() (I don not understand this logic??)", func(t *testing.T) {
```

Comment:
I forget to remove this comment. I and my teammate discuss about do we need to lookup regardless of region. I think this can help if we don't really focus on pricing as regional level. I gonna remove this comment.
https://github.com/opencost/opencost/blob/ebdd959fbe20f73008ba1b83ce449184ad1754ee/pkg/cloud/provider/csvprovider.go#L239-L247